Verify and use pod constants returned in the SetupPod response

itsmojo opened the pull request:
Enforce 0.05U pulse size must mandatorily match as per PDM
Verify secondsPerBolusPulse & secondsPerPrimePulse are 2 & 1
Verify that reported pod lifetime matches expected 80 hours
Adjust prime and cannula insertion bolus sizes to pod values
Updated code & comments for setup bolusing to correctly scale
Updated comments for pulse unit sizing is for U100 insulin
Update pairing RSSI message and fix a string format error
Fix to properly handle activation timeout error w/o a pod fault

itsmojo committed:
Verify and use pod constants returned in the SetupPod response
Enforce 0.05U pulse size must mandatorily match as per PDM
Verify secondsPerBolusPulse & secondsPerPrimePulse are 2 & 1
Verify that reported pod lifetime matches expected 80 hours
Adjust prime and cannula insertion bolus sizes to pod values
Updated code & comments for setup bolusing to correctly scale
Updated comments for pulse unit sizing is for U100 insulin
Update pairing RSSI message and fix a string format error
Fix to properly handle activation timeout error w/o a pod fault

itsmojo committed:
Updated testParsingLongVersionResponse to test new returned values

itsmojo committed:
Comment updates to match current 01 Version Response wiki

ps2 commented on OmniKit/Model/Pod.swift:
```diff
@@ -57,13 +57,15 @@ public struct Pod {
-    public static let primeUnits = 2.6
+    // Default amount for priming bolus using secondsPerPrimePulse timing.
+    // Can potentially be overrided with value returned by pod during the pairing process.
+    public static var primeUnits = 2.6
```
These should be constants. There should be no modifiable state here.

itsmojo replied:
Changed

ps2 reviewed (changes requested):
Thanks for the updated decoding!  I think generally this looks good, except for the storing of state on the Pod struct. If we really need to track this as state (not sure that's really needed), then it should go on the PodState or PumpManager State.

itsmojo committed:
Verify prime & cannula insertion bolus sizes match Pod constants
Update Pod comments to note all the verified fields
Update VersionResponse.debugDescription to include all fields
Update MessageTest to verify all VersionResponse fields
Improve VersionResponse comments for various cases
Fix typo in the pairingCompleted PodProgressStatus string

itsmojo committed:
Reworked logic to cleanly deactivate for pods with incompatible constants
Fixed UI error handling bugs resulting from switch to PumpManagerErrors
Created new PodCommsError error for an incompatible pod
Created new PodCommsError.isFaulted var to check for all fault types
Created new SetupProgress value to mark PodState as incompatible
Updated PodState isFaulted var to include new podIncompatible check
Display all incompatible values instead of just the first one found

itsmojo committed:
Minor OmnipodPumpManager error tweaks for consistency
